src/routers/alpha-router/gas-models/v3/v3-heuristic-gas-model.ts

```diff
@@ -259,6 +269,81 @@ export class V3HeuristicGasModelFactory extends IOnChainGasModelFactory {
         throw err;
       }
 
+      /*
+        We want to get the gas cost in terms of the quote token.
+        Sometimes, the quote token / native token pair is very illiquid and the price is incorrect.
+        We also have access to the input token / native token pair, which is more likely to be liquid.
```

Comment:
hmm yeah you're right, I was thinking that it is likely that at least one of the native pools is liquid, otherwise it would be a direct swap between two illiquid coins and we would fail to give a good quote anyways.